### ISSUE TITLE
Superapp cleanup

### ISSUE THREAD

**Issue by kandrio** [Tribler/trustchain-superapp#161]
After our discussion with @synctext (https://github.com/Tribler/tribler/issues/7438#issuecomment-1632366938), I'm creating this "parent" issue to describe what needs to be done in the `trustchain-superapp` in order to make it lighter and more efficient.

Here's a list of issues we need to tackle:
1. Keep only 6 sub-modules:
    - `musicdao`: https://github.com/Tribler/trustchain-superapp/tree/master/musicdao
    - `Peer AI` (?)
    - `freedomOfComputing`: https://github.com/Tribler/trustchain-superapp/tree/master/freedomOfComputing
    - `IDelft` (?)
    - `eurotoken`: https://github.com/Tribler/trustchain-superapp/tree/master/eurotoken
    - `debug`: https://github.com/Tribler/trustchain-superapp/tree/master/debug
2. Ensure that, each time the superapp runs, only the needed module gets executed, and there are no unnecessary modules loaded or running in the background.
3. Make `musicdao` a mature prototype by:
    - improving coverage and testing (https://developer.android.com/training/testing/espresso)
    - fixing the GUI
    - improving code readability
    - integrating the recommendation tab: https://github.com/Tribler/trustchain-superapp/pull/160
4. Add/fix Github actions

**Comment by kandrio** [comment]
Today, I had a meeting with @InvictusRMC and we created a skeleton for my work in the upcoming weeks:
1. *_Independence_*: Make the six modules that I mentioned in the previous comment *completely independent* (from one another and of other modules): Currently some of these modules import classes from other modules, so most probably I will need to move these class definitions into a common-code directory.
2. *_Cleanup_*: Drop all other modules apart from these six.
3. *_Testing_*: Use the `espresso` library (https://developer.android.com/training/testing/espresso) for e2e testing on `musicdao`. My work will be a baseline for testing for future projects that students will work on in the next years.

**Comment by synctext** [comment]
Solid plans! Please focus on a Google Play Store and live demo for 5 Sep @ TUDelft campus. So ask @InvictusRMC for a 1 Sep .APK upload for review to Google or obtain your own account from him.

For market leaders such as GoldRepublic it would be great to offer "GoldEuro" coins within our Superapp. So the demo is for offline token transfer. Web-of-Trust is great concept to explain. https://github.com/Tribler/trustchain-superapp#digital-euro  _Future work_: we need a simple understandable rule around offline double spending. We re-define it as "overspending" with negative balance tied to your passport-level digital identity. Irrefutable, you can't escape your fraud. Second, "first-to-claim" rule. When coming online again the ordering is determined of the claims on an "overspender". This is the order the system follows on repayment of accumulated debt by overspender.

**Comment by kandrio** [comment]
# MusicDAO Independence

`MusicDAO` depends from:
1. `currencyii`

**Comment by kandrio** [comment]
# IDelft (`valuetransfer`) Independence

`IDelft` depends from the following modules:
1. `peerchat` : https://github.com/Tribler/trustchain-superapp/tree/master/peerchat

**Comment by kandrio** [comment]
# Pruning of old modules

## Local Environment

I've successfully pruned the following modules in my local environment:
- [X] `DeToks`
- [X] `LiteratureDAO`
- [X] `AtomicSwap`
- [X] `TrustchainPayloadGenerator`
- [X] `TrustchainTrader`
- [ ] `TrustchainExplorer`
- [ ] `TrustchainVoter`
- [ ] `LiquidityPool`

## Upstream

I've successfully pruned from our GitHub repo the following modules:
- [ ] `DeToks`
- [ ] `LiteratureDAO`
- [ ] `AtomicSwap`
- [ ] `TrustchainPayloadGenerator`
- [ ] `TrustchainTrader`
- [ ] `TrustchainExplorer`
- [ ] `TrustchainVoter`
- [ ] `LiquidityPool`

**Comment by kandrio** [comment]
# Trustchain Explorer

As specified in the comment above (https://github.com/Tribler/trustchain-superapp/issues/161#issuecomment-1645454443), the Trustchain Explorer is one of the modules we need to get rid of. However, there is one issue: the parent app invokes the `TrustChainExplorerActivity` Activity whenever the user clicks on the notification. Here's a more detailed explanation of the issue...

## Issue with Trustchain Explorer

When the user installs the Trustchain Superapp on their device, there is a **notification** that looks like this:

<img src="https://github.com/Tribler/trustchain-superapp/assets/49658958/cbcd1330-25a4-435b-9f75-914647c63862" width="30%" />

When clicking on that notification, the Trustchain Explorer module opens up:

<img src="https://github.com/Tribler/trustchain-superapp/assets/49658958/0fa52339-e94c-4e7a-8286-028f888ca90c" width="30%" />



## Decisions to make

**Comment by synctext** [comment]
agreed! just het rid of these notifications.

**Comment by kandrio** [comment]
@synctext I looked into dropping the notification entirely, but I found that we can't do it through the `trustchain-superapp`. Take a look at the `UPDATE: Feasible Solution` section of the above comment.

**Comment by synctext** [comment]
good idea. Smart, keep IPv8 unpatched. please also modify the notification into something sensible, like, superapp running.

**Comment by kandrio** [comment]
@InvictusRMC I created a PR where I prune all the [old modules](https://github.com/Tribler/trustchain-superapp/issues/161#issuecomment-1645454443). These are the modules that are completely useless to the ones that we plan to keep. It will help to get rid of these first, and then to move to the remaining modules that need to be pruned. 

The five modules that we plan to keep still depend on the following remaining modules: `currencyii`, `peerchat`, `ig-ssi`. I won't prune these in this PR. However, I will get rid of these three in my next PR.

**Comment by kandrio** [comment]
@InvictusRMC After the above PR that prunes 9 unwanted modules, I created another PR that prunes `currencyii`. The latter was harder since `musicdao` (which we plan to keep) heavily depended on `currencyii`. Hence, I created a standalone PR just for the prunning of `currencyii`.

CC @synctext

**Comment by kandrio** [comment]
@InvictusRMC I'm continuing the effort here. I created another PR (https://github.com/Tribler/trustchain-superapp/pull/164) that prunes `ig-ssi`. This one was a complex module to prune as well because `iDelft` depended on it.

**Comment by synctext** [comment]
perfect effort people!
Please plan toward a Google Playstore upload of a fully operational .apk
That process may take 7-10 days.

**Comment by kandrio** [comment]
Yet another PR here:
- https://github.com/Tribler/trustchain-superapp/pull/165

This PR prunes the last modules in our list:
- `gossipML`
- `Peerchat`

Peerchat in particular was very difficult to prune as it was used not only by `iDelft` but also by the entire parent `app`.

This PR should conclude the effort of pruning all of the unnecessary modules. For a summary of what modules I pruned in total, take a look at this comment:
- https://github.com/Tribler/trustchain-superapp/issues/161#issuecomment-1645454443

Here's a summary of the PRs that I've created (should be merged in that order):
1. https://github.com/Tribler/trustchain-superapp/pull/162
2. https://github.com/Tribler/trustchain-superapp/pull/163
3. https://github.com/Tribler/trustchain-superapp/pull/164
4. https://github.com/Tribler/trustchain-superapp/pull/165

CC @InvictusRMC @synctext

**Comment by kandrio** [comment]
# Current State
We have:
- pruned a number of old modules (branch: `core`)
- uploaded a Play Store release
- created a PR (https://github.com/Tribler/trustchain-superapp/pull/167) to merge `core` into `master` (@InvictusRMC let me know if I can help with merging this)

# Next Steps
Looking at the original list of tasks (https://github.com/Tribler/trustchain-superapp/issues/161#issue-1807532878), I see we have the following things remaining:
1. `PeerAI` is not part of `Tribler/trustchain-superapp`. Instead, it currently lives in @quintene's repo: https://github.com/quintene/trustchain-superapp. We should create a PR and merge that to `Tribler/trustchain-superapp`.
2. Make `MusicDAO` a mature prototype:
    - **_Review and merge the recommendation tab_** (relevant PR: https://github.com/Tribler/trustchain-superapp/pull/160).
    - **_Introduce testing_** ([espresso](https://developer.android.com/training/testing/espresso) library). Judging from the contents of the [musicdao/src/test](https://github.com/Tribler/trustchain-superapp/tree/master/musicdao/src/test/resources) directory, there are no tests currently. We should **start with unit tests** and then add integration tests and e2e tests.
    - **_Fix the GUI_**: I don't know what that refers to. Are there specific problems with the GUI?
    - **_Improve code readability_**.
3. Add/fix GitHub actions: I don't think that GitHub Actions are broken. Regarding adding more, any ideas?

@synctext @InvictusRMC any comments?

**Comment by InvictusRMC** [comment]
My ideas:

- PeerAI: Should indeed be merged. Feel free to open a PR.
- MusicDAO: Another MSc student is currently working on MusicDAO. Would let this be for now.
- GUI tests with scenarios would be awesome!

Issues I am aware of:

- Technical debt: tons of deprecation warnings during building. These should all be addressed at one point
- Older APIs: we have specific logic for certain Android APIs. In some instances (e.g., permissions) this is required; in other instances, we simply perform a new action in case an older API is used. We can raise the minimum API, or fix the behavior for older APIs
- BT: To my knowledge, currently one cannot find peers using Bluetooth unless using a very old Android version. To my understanding, the BT permissions have not been properly handled ever since Android changed it. As such, it does not function on new devices.

**Comment by kandrio** [comment]
@InvictusRMC Thanks a lot for the update. I'm currently really busy with finishing my literature survey (https://github.com/Tribler/tribler/issues/7438#issuecomment-1673453795). So, I'll focus on the superapp towards the end of the week.

First things I plan to kill are:
- merging `peerAI`
- fixing the camera bug for IDelft Chat #168 
- raise the minimum API
